data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AbstractDataService.java

```diff
@@ -0,0 +1,238 @@
+        this.auditService = auditService;
+
+        LOGGER.debug("Initialised {} with readers {} and serialisers {}", this.getClass(), this.readers, this.serialisers);
+    }
+
+    @Generated
+    public Collection<DataReader> getReaders() {
+        return readers;
+    }
+
+    @Generated
+    public void setReaders(final Collection<DataReader> readers) {
+        this.readers = Optional.ofNullable(readers)
+                .orElseThrow(() -> new IllegalArgumentException("readers cannot be null"));
+    }
+
+    @Generated
+    public Map<String, Class<Serialiser<?>>> getSerialisers() {
+        return serialisers;
+    }
+
+    @Generated
+    public void setSerialisers(final Map<String, Class<Serialiser<?>>> serialisers) {
+        this.serialisers = Optional.ofNullable(serialisers)
+                .orElseThrow(() -> new IllegalArgumentException("serialisers cannot be null"));
+    }
+
+    /**
+     * Implement a transformation from reader/serialiser/rule output as a callback for the last point of auditing.
+     *
+     * @return a flow from reader/serialiser/rule output bytes to the output of {@link AbstractDataService#defaultRunnableGraph(AuditableAuthorisedDataRequest)}
+     * which allows for injecting 'interesting' behaviour before auditing.
+     */
+    protected abstract Flow<ByteString, ByteString, NotUsed> transformResponse();
+
+    // Suppress warning for untyped lambda expressions, where the type is not known until execution time
+    @SuppressWarnings("java:S2211")
+    protected Source<ByteString, CompletionStage<Done>> defaultRunnableGraph(final AuditableAuthorisedDataRequest auditable) {
+        AbstractDataService service = this;
+        LOGGER.info("Selected service '{}' based on route path", service.getClass());
+
+        final AtomicLong recordsProcessed = new AtomicLong(0);
+        final AtomicLong recordsReturned = new AtomicLong(0);
+
+        var authorisation = Optional.ofNullable(auditable.getAuthorisedDataRequest());
+        LOGGER.debug("Authorisation was {}, but effect will be delegated until stream materialisation", authorisation.isPresent());
+
+        var responseSource = Source
+                // Construct lazily so errors in construction fall through with errors in reading
+                .lazySource(() -> {
+                    AuthorisedDataRequest authorised = authorisation
+                            .orElseThrow(() -> new ForbiddenException("Authorisation denied for request " + auditable.getDataRequest(), auditable.getAuditErrorMessage().getError()));
+                    LOGGER.debug("Request '{}' was authorised", auditable.getDataRequest());
+
+                    User user = authorised.getUser();
+                    LeafResource leafResource = authorised.getResource();
+                    Context context = authorised.getContext();
+                    @SuppressWarnings("unchecked") Rules<Serializable> rules = (Rules<Serializable>) authorised.getRules();
```

Comment:
can this be split over 2 lines?
```suggestion
                    @SuppressWarnings("unchecked") 
                    Rules<Serializable> rules = (Rules<Serializable>) authorised.getRules();
```